File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[]) {
    //check arguments
    if(argc != 2) {
        printf("Invalid number of arguments");
        return 1;
    };
    //ask user for input
    printf("plaintext: ");
    string s = get_string();
    if (s != NULL) {
        printf("ciphertext: ");
        //calculate key
        int k = atoi(argv[1]) % 26;
        for (int i = 0, n = strlen(s); i < n; i++) {
            //operate on uppercase letters
            if((s[i] >= 'A' && s[i] <= 'Z')) {
                printf("%c", (char) ((s[i] -'A' + k)%26+'A')); 
            }
            //operate on lowercase letters
            else if(s[i] >= 'a' && s[i] <= 'z') {
                printf("%c", (char) ((s[i] -'a' + k)%26+'a'));
            } else {
                //reprint non alphabetic letter
                printf("%c", (char) s[i]);
            }
        };
    };
    printf("\n");
    return 0;
};

File: pset6/caesar/caesar.py
from cs50 import get_string
import sys


def main():
    if len(sys.argv) is not 2:
        print('Invalid number of arguments')
        exit(1)
    print('plaintext: ', end='')
    # ask for input
    string = get_string()
    if string is not None:
        print('ciphertext: ', end='')
        # print('ciphertext: {}'.format(k))
        # calculate key
        k = int(sys.argv[1]) % 26
        for i in string:
            # check for upper case letter
            if i.isupper() is True:
                newChar = chr((ord(i) - ord('A') + k) % 26 + ord('A'))
                print(newChar, end='')
            # check for lower case letter
            elif i.islower() is True:
                newChar = chr((ord(i) - ord('a') + k) % 26 + ord('a'))
                print(newChar, end='')
            # reprint non letter character
            else:
                print(i, end='')
        print()
        exit(0)

if __name__ == '__main__':
    main()
